homeassistant/components/ezviz/__init__.py

```diff
@@ -50,69 +55,71 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
         hass.config_entries.async_update_entry(entry, options=options)
 
-    if entry.data.get(CONF_TYPE) == ATTR_TYPE_CAMERA:
-        if hass.data.get(DOMAIN):
-            # Should only execute on addition of new camera entry.
-            # Fetch Entry id of main account and reload it.
-            for item in hass.config_entries.async_entries():
-                if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
-                    _LOGGER.info("Reload EZVIZ integration with new camera rtsp entry")
-                    await hass.config_entries.async_reload(item.entry_id)
+    # Initialize EZVIZ cloud entities
+    if PLATFORMS_BY_TYPE[sensor_type]:
+        # Initiate reauth config flow if account token if not present.
+        if not entry.data.get(CONF_SESSION_ID):
+            raise ConfigEntryAuthFailed
+
+        ezviz_client = EzvizClient(
+            token={
+                CONF_SESSION_ID: entry.data.get(CONF_SESSION_ID),
+                CONF_RFSESSION_ID: entry.data.get(CONF_RFSESSION_ID),
+                "api_url": entry.data.get(CONF_URL),
+            },
+            timeout=entry.options.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        try:
+            await hass.async_add_executor_job(ezviz_client.login)
 
-        return True
+        except (EzvizAuthTokenExpired, EzvizAuthVerificationCode) as error:
+            raise ConfigEntryAuthFailed from error
 
-    try:
-        ezviz_client = await hass.async_add_executor_job(
-            _get_ezviz_client_instance, entry
+        except (InvalidURL, HTTPError, PyEzvizError) as error:
+            _LOGGER.error("Unable to connect to Ezviz service: %s", str(error))
```

Comment:
Don't log and raise. It will spam the log on retry. Just pass the log message as an exception argument to `ConfigEntryNotReady`.

Reply:
Addressed in https://github.com/home-assistant/core/pull/94384